### PR TITLE
Implemented label for 3D-printer marker

### PR DESCRIPTION
**Pull request by InMyOrbit**
Note that it is unknown where the printer cuts the label, this implementation might have to be adjusted after testing it on a printer.

### CODE REVIEW

**Comment by emanuelen5** [comment]
Don't fotgängare typ generate an image of the label in the CI scripts (by adding it to the file .github/workflows/pythonapp.yml)

**Review by emanuelen5 — APPROVED** [review]

**Comment by emanuelen5** [comment]
I haven't tested it though

**Comment by emanuelen5** [comment]
I have now tested it and it works fine! It would also be nice to add a regression test that makes sure that the dimensions of the label stays as intended.

**Comment by emanuelen5** [comment]
- [ ] Double check dimensions of label
- [ ] Add regression test for label size
- [ ] Add github workflow image upload of example label

**Comment by emanuelen5** [comment]
I added the Github workflow in the commit above ^

**Comment by InMyOrbit** [comment]
> I have now tested it and it works fine! It would also be nice to add a regression test that makes sure that the dimensions of the label stays as intended.

Added unit test for size. Should we enhance test by creating X random strings representing names and test with them as well?

**Comment by InMyOrbit** [comment]
We still have to test this on the actual printer, we have not yet confirmed that the height is actually 25 mm.

File: src/label/creator.py [hunk 1]
```diff
@@ -271,6 +281,10 @@ def get_font_size_estimation(text):
     return get_font_size_estimation_from_lookup_table(len(text))
 
 
+def get_label_height_in_px(label_height_mm):
```
Comment (emanuelen5) [inline src/label/creator.py:284]: Does this function only work for the height (the name implies it)? I suspect it might work for width also, and thus can be renamed to `get_mm_in_px()`/`label_mm_to_px()` or similar.

Comment (InMyOrbit) [reply src/label/creator.py:284]: You are right, but width is hard-coded, might have a use for positioning of stuff in the future though.

File: tests/test_labels.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import unittest
+import math
+from src.label.creator import create_3d_printer_label, IMG_WIDTH
+
+MEMBER_ID = "1140"
+MEMBER_NAME = "Firstname Lastname"
```
[the remark this answers is not in the document]
Comment (InMyOrbit) [reply tests/test_labels.py:6]: Let's add test cases for extremes. Short name, 5 random characters? Long name, 40 random characters?

File: src/label/creator.py [hunk 3]
```diff
@@ -16,7 +16,12 @@
...
 QR_CODE_DESCRIPTION_MAX_LENGTH = 100
 
-IMG_WIDTH = 696  # From brother_ql for 62 mm labels
+# For Brother QL-810W with 62 mm wide labels
+PRINTER_HEIGHT_MARGIN_MM = 3
+PRINTER_PIXELS_PER_MM = 300 / 25.4
+PRINTER_LABEL_PRINTABLE_WIDTH = 58
+
+IMG_WIDTH = math.floor(PRINTER_PIXELS_PER_MM * PRINTER_LABEL_PRINTABLE_WIDTH)
 IMG_HEIGHT = math.floor((58 + 20) / 25.4 * 300)
```
Comment (emanuelen5) [inline src/label/creator.py:25]:
```suggestion
IMG_HEIGHT = math.floor((58 + 20) * PRINTER_PIXELS_PER_MM)
```

File: src/label/creator.py [hunk 4]
```diff
@@ -119,12 +127,18 @@ def __init__(self, image, label_width=CANVAS_WIDTH):
...
-    def __init__(self, label_objects):
+    def __init__(self, label_objects, label_height_mm=None):
 
         self.label_width = CANVAS_WIDTH
         self.label_objects = label_objects
 
-        self.label_height = self.get_canvas_height() + ((len(self.label_objects) + 1) * IMG_MARGIN)
+        if label_height_mm is None:
+            self.label_margin = IMG_MARGIN
+            self.label_height = self.get_canvas_height() + ((len(self.label_objects) + 1) * self.label_margin)
```
Comment (emanuelen5) [inline src/label/creator.py:137]: It's a bit hard to follow what units each variable is expressed in. For example, are these in pixels?

**Review by emanuelen5 — APPROVED** [review]
It works, so no need to change, but some suggestions

File: src/label/creator.py (hunk 3 above)
Comment (InMyOrbit) [reply src/label/creator.py:25]: Not the same thing, this is connected to the first labels developed, didn't dare too touch them without going thru what the effects would be.

File: src/label/creator.py (hunk 4 above)
Comment (InMyOrbit) [reply src/label/creator.py:137]: I do agree, but that refactoring should be another issue.

**Comment by InMyOrbit** [comment]
I do agree with what you pointed out, but it is not directly related to this change so I suggest that we make it another issue instead.
